### PR TITLE
Add OpenCL variants on MacOS

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

## Checks

- [ ] If unspecified, default package should be `opencl_impl == apple`
- [ ] When installing another package that uses OpenCL on the same env, Conda should install the variant `opencl_impl == khronos` (for new and existing environments)
- [ ] User can select variants using the build string `openmm=*=*apple*` or `openmm=*=*khronos*`

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,7 +13,8 @@ source:
 
 build:
   number: {{ build }}
-  skip: true  # [((cuda_compiler_version == "None") and (linux64 or win)) or ((cuda_compiler_version != "10.2") and (ppc64le))]
+  string: "py{{ PY_VER }}h{{ PKG_HASH }}_{{ build }}_{{ opencl_impl }}"  # [osx]
```
Comment (isuruf) [inline recipe/meta.yaml:16]: To make khronos low priority
```suggestion
  string: "py{{ PY_VER }}h{{ PKG_HASH }}_{{ build }}_{{ opencl_impl }}"  # [osx]
  track_features:
    - openmm_opencl_khronos  # [opencl_impl == 'khronos']
```

Comment (jaimergp) [reply recipe/meta.yaml:16]: Oh, nice! Thanks!

**Comment by peastman** [comment]
As far as I understand it (which is not very far!), it looks ok to me.

**Review by peastman — APPROVED** [review]

**Comment by jaimergp** [comment]
Ok! I am going to merge to the test label and run some checks whether the behavior is the one we expect. I'll post the instructions here so MacOS users can help out!

**Comment by jaimergp** [comment]
Check https://github.com/conda-forge/openmm-feedstock/issues/21#issuecomment-735855671 for instructions!

**Pull request merged**